lib/src/util/dart_type_utilities.dart

```diff
@@ -288,8 +288,16 @@ extension on TypeSystem {
       // Otherwise, they might be related.
       return false;
     } else {
+      var sameSupertypes = leftElement.supertype == rightElement.supertype;
+
+      // Unrelated Enums or Mixins have the same supertype, but they are not the same element, so
+      // they are unrelated.
+      if (true && (leftElement is EnumElement || leftElement is MixinElement)) {
```

Comment:
`true &&`? That's curious.

Reply:
Oh, nice catch. It remained while debugging. It used to be `sameSupertypes && (leftElement ...)` but it doesn't seem to be required. Should we add it anyway? I don't have enough background with the linter, so there may be edge cases I'm not aware of.

Reply:
Yeah I guess you can revert the `sameSupertypes` calculation, and keep that at the end.